tests/bluetooth/tester/src/btp_bap.c

```diff
@@ -17,35 +18,1258 @@
+		      const struct bt_codec *codec, struct bt_bap_stream **stream,
+		      struct bt_codec_qos_pref *const pref, struct bt_bap_ascs_rsp *rsp)
+{
+	struct audio_connection *audio_conn;
+	struct audio_stream *stream_wrap;
+
+	LOG_DBG("ASE Codec Config: ep %p dir %u", ep, dir);
+
+	print_codec(codec);
+
+	if (validate_codec_parameters(codec)) {
+		*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_CONF_REJECTED,
+				       BT_BAP_ASCS_REASON_CODEC_DATA);
+
+		btp_send_ascs_operation_completed_ev(conn, ep->status.id, BT_ASCS_CONFIG_OP, 1);
+
+		return -ENOTSUP;
+	}
+
+	audio_conn = find_connection(conn);
+	stream_wrap = stream_alloc(audio_conn);
+
+	if (stream_wrap == NULL) {
+		LOG_DBG("No free stream available");
+		*rsp = BT_BAP_ASCS_RSP(BT_BAP_ASCS_RSP_CODE_NO_MEM, BT_BAP_ASCS_REASON_NONE);
+
+		btp_send_ascs_operation_completed_ev(conn, ep->status.id, BT_ASCS_CONFIG_OP, 1);
+
+		return -ENOMEM;
+	}
+
+	*stream = &stream_wrap->stream;
+	LOG_DBG("ASE Codec Config stream %p", *stream);
+
+	if (dir == BT_AUDIO_DIR_SOURCE) {
+		audio_conn->configured_source_stream_count++;
+	} else {
+		audio_conn->configured_sink_stream_count++;
+	}
+
+	*pref = qos_pref;
+
+	btp_send_ascs_operation_completed_ev(conn, ep->status.id, BT_ASCS_CONFIG_OP, 0);
+
+	return 0;
+}
+
+static int lc3_reconfig(struct bt_bap_stream *stream, enum bt_audio_dir dir,
+			const struct bt_codec *codec, struct bt_codec_qos_pref *const pref,
+			struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("ASE Codec Reconfig: stream %p", stream);
+
+	print_codec(codec);
+
+	btp_send_ascs_operation_completed_ev(stream->conn, stream->ep->status.id,
+					     BT_ASCS_CONFIG_OP, 0);
+
+	return 0;
+}
+
+static int lc3_qos(struct bt_bap_stream *stream, const struct bt_codec_qos *qos,
+		   struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("QoS: stream %p qos %p", stream, qos);
+
+	print_qos(qos);
+
+	btp_send_ascs_operation_completed_ev(stream->conn, stream->ep->status.id,
+					     BT_ASCS_QOS_OP, 0);
+
+	return 0;
+}
+
+static int lc3_enable(struct bt_bap_stream *stream, const struct bt_codec_data *meta,
+		      size_t meta_count, struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("Enable: stream %p meta_count %zu", stream, meta_count);
+	btp_send_ascs_operation_completed_ev(stream->conn, stream->ep->status.id,
+					     BT_ASCS_ENABLE_OP, 0);
+
+	return 0;
+}
+
+static int lc3_start(struct bt_bap_stream *stream, struct bt_bap_ascs_rsp *rsp)
+{
+	LOG_DBG("Start: stream %p", stream);
+	btp_send_ascs_operation_completed_ev(stream->conn, stream->ep->status.id,
+					     BT_ASCS_START_OP, 0);
+
+	return 0;
+}
+
+static bool valid_metadata_type(uint8_t type, uint8_t len, const uint8_t *data)
+{
+	switch (type) {
+	case BT_AUDIO_METADATA_TYPE_PREF_CONTEXT:
+	case BT_AUDIO_METADATA_TYPE_STREAM_CONTEXT:
+		if (len != 2) {
+			return false;
+		}
+
+		/* PTS wants us to reject the parameter if reserved bits are set */
+		if ((data[1] & 0xf0) > 0) {
```

Comment:
Suggest to get the context as a 16-bit value and compare to `BT_AUDIO_CONTEXT_TYPE_ANY`

The metadata values below (language, rating, program_info etc.) also have defined values; should we not verify whether or not those are RFU too?